subprojects/core/src/main/java/org/gradle/api/internal/file/delete/DefaultDeleteSpec.java

```diff
@@ -17,15 +17,19 @@
 package org.gradle.api.internal.file.delete;
 
 import org.gradle.api.NonExtensible;
+import org.gradle.api.NonNullApi;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Property;
 
 @NonExtensible
+@NonNullApi
 public class DefaultDeleteSpec implements DeleteSpecInternal {
     private Object[] paths;
-    private boolean followSymlinks;
+    private final Property<Boolean> followSymlinksProp;
```

Comment:
You should be able to make `DefaultDeleteSpec` abstract and change `DefaultFileOperations.delete` to instantiate it via ObjectFactory.newInstance.  

That will get rid of this boilerplate (no field, only an abstract getter), but it's also necessary to make this work in Groovy DSL.

Otherwise, this will not work:
```
delete {
   followSymlinks = true
   ...
}
```

Reply:
> Otherwise, this will not work

This worked with the current state (the tests were green). However, simplifying that is great. I've updated the PR, thanks